README.md

```diff
@@ -1,2 +1,152 @@
+* `executor` and `dev_handler` are connected via `shm_wrapper`. This connection is used for `executor` to send commands to the attached devices (tell a motor to run at a certain speed, move a servo to certain position, etc.), and for `dev_handler` to serve `executor` with the device data that it needs to run the student code.
+* `dev_handler` communicates with `lowcar` via serial connection. This connection is used to poll for new devices, detect when devices have disconnected, and send device data and commands between the Raspberry Pi and the Arduinos.
+* All three processes output log messages through the `logger` tool. If logs are to be sent over the network to Dawn, those logs are put into a FIFO pipe which is opened by `net_handler`, where the logs are processed and sent to Dawn.
+
+## Contents
+
+The following details the folders in the repository:
+
+* `net_handler`: contains routines for managing the TCP and UDP connections to Dawn and Shepherd, the main routine for managing the managers of the connections, the protobuf definitions and compiled protobufs, utility functions specific to `net_handler`, test files, and documentation.
+* `dev_handler`: contains routines for managing the connections between the Raspberry Pi and the devices, the main routine for managing the managers of the connections, utilities for unpacking and packing messages to be sent to and from the devices, test files, and documentation.
+* `executor`: contains routines for spawning new processes to run the student code (written in Python) in the various modes (autonomous, teleop, coding challenge), an implementation of the StudentAPI (the functions that PiE provides for students to interact with the devices and gamepad) in Cython, the student code itself, test files, and documentation.
+* `shm_wrapper` and `shm_wrapper_aux`: contains the implementations of the tools described in the previous section, test files for both wrappers, a simple program in each folder to probe the state of the corresponding shared memory blocks while Runtime is running, and documentation.
+* `logger`: contains the implementation of the tool described in the previous section, a configuration file, test files, and documentation.
+* `runtime_util`: contains general utility functions that are useful to all of the processes, mostly dealing with getting information about device types and their parameters. The header file defines many important constants that are used throughout Runtime.
+* `systemd-stuff`: contains the services that are put into `systemd` that start Runtime on bootup of the Raspberry Pi.
+
+## Dependencies and Concepts
+
+Remember, one of the design principles is KISS. Nevertheless, Runtime is a complex system and will require effort to learn and understand. Hopefully this list does not seem too daunting to you! <sup id="return3">[3](#footnote3)</sup>
+
```

Comment:
You should also specify the dependencies of POSIX(Linux highly recommended), C, gcc, Python >= 3.7 (3.6 might also work), Python headers.

Reply:
👌